### DISCUSSION TITLE
Markdown Editor Preview has no line breaks

### DISCUSSION THREAD

**Discussion by khanakia** [filamentphp/filament#11266] [Help]
### Package

Form builder

### Package Version

^3.0-stable

### How can we help you?

I am using below form code and when I view the markdown it does not adds any line breaks

```php
Forms\Components\MarkdownEditor::make('body')
                    ->columnSpanFull()
```

Acutal Code
```
### projects Table
`id`
`sem_id` - this is the projectid returned by semrush
`ref` - string (This will save the client or app name made the request let say `risify`)
```

This is how Form View looks it does not adds any line breaks
```
projects Table
`id` `sem_id` - this is the projectid returned by semrush `ref` - string (This will save the client or app name made the request let say risify) 
```

![Screenshot 2024-02-06 at 1 19 49 PM](https://github.com/filamentphp/filament/assets/2174170/5972a1f5-5d75-42dd-b463-36fa2d3c0ea4)

![Screenshot 2024-02-06 at 1 19 32 PM](https://github.com/filamentphp/filament/assets/2174170/942f282b-d674-4d2c-80ca-d25dd6b36be7)

**Comment by leandrocfe** [comment]
try to add a new line break

![image](https://github.com/filamentphp/filament/assets/3833889/909bb3dc-b5c7-4bc9-8400-d2ab50556297)


![image](https://github.com/filamentphp/filament/assets/3833889/16a6ad2e-903e-4d65-a2bf-2267cda6695b)

**Reply by khanakia** [reply]
Yes, i know this. But why line breaks cannot be the default. If you check the Github, stackedit, obsidian all other markdown editors they have line breaks by default.

**Reply by leandrocfe** [reply]
Di you try an infolist markdown?
https://filamentphp.com/docs/3.x/infolists/entries/text#rendering-markdown-as-html

**Reply by khanakia** [reply]
How do i use this with forms ? I have below code
```
Forms\Components\MarkdownEditor::make('body')
                    ->columnSpanFull()
```

**Reply by leandrocfe** [reply]
https://filamentphp.com/docs/3.x/panels/resources/viewing-records#using-an-infolist-instead-of-a-disabled-form

**Reply by khanakia** [reply]
Infolist has same issue it just looks the same and has no line breaks
![Screenshot 2024-02-06 at 3 11 47 PM](https://github.com/filamentphp/filament/assets/2174170/2a60508a-65ef-4467-a2ac-5ec50e149e4c)

**Comment by jinbatsu** [comment]
Have the same problem, how to line breaks in view infolist, and in table list textcolumn.

in Infolist:
```php
TextEntry::make('body')  
  ->markdown()
```

in Table Column:
```php
TextColumn::make('body')  
  ->markdown()
```

Well, ok, **solved** using this, both infolist and in table list column:
```php
TextEntry::make('body')
  ->formatStateUsing(fn($state) => new HtmlString(nl2br($state)))
  ->markdown()
```
Line breaks works, and markdown applied.
